AEPServices/Sources/ui/fullscreen/FullscreenMessage.swift

```diff
@@ -11,43 +11,204 @@
+
+                // Fix for iPhone X to display content edge-to-edge
+                if #available(iOS 11, *) {
+                    wkWebView.scrollView.contentInsetAdjustmentBehavior = .never
+                }
+
+                // save the HTML payload to a local file if the cached image is being used
+                var useTempHTML = false
+                var cacheFolderURL: URL?
+                var tempHTMLFile: URL?
+                let cacheFolder: URL? = self.fileManager.getCacheDirectoryPath()
+                if cacheFolder != nil {
+                    cacheFolderURL = cacheFolder?.appendingPathComponent(self.DOWNLOAD_CACHE)
+                    tempHTMLFile = cacheFolderURL?.appendingPathComponent(self.TEMP_FILE_NAME).appendingPathExtension(self.HTML_EXTENSION)
+                    if !self.isLocalImageUsed {
+                        /* AMSDK-8942: The ACS extension saves downloaded remote image files in the cache. We have to use loadFileURL so we can allow read access to these image files in the cache but loadFileURL expects a file URL and not the string representation of the HTML payload. As a workaround, we can write the payload string to a temporary HTML file located at cachePath/adbdownloadcache/temp.html and pass that file URL to loadFileURL.
+                         */
+                        do {
+                            try FileManager.default.createDirectory(atPath: cacheFolderURL!.path, withIntermediateDirectories: true, attributes: nil)
+                            try self.payload.write(toFile: tempHTMLFile!.path, atomically: true, encoding: .utf8)
+                            useTempHTML = true
+                        } catch {
+                            Log.debug(label: self.LOG_PREFIX, "Failed to save the temporary HTML file for fullscreen message \(error)")
+                        }
+                    }
+                }
+                // load the HTML string on WKWebview. If we are using the cached images, then use
+                // loadFileURL:allowingReadAccessToURL: to load the html from local file, which will give us the correct
+                // permission to read cached files
+                if useTempHTML {
+                    wkWebView.loadFileURL(URL.init(fileURLWithPath: tempHTMLFile?.path ?? ""), allowingReadAccessTo: URL.init(fileURLWithPath: cacheFolder?.path ?? ""))
+                } else {
+                    wkWebView.loadHTMLString(self.payload, baseURL: Bundle.main.bundleURL)
+                }
+
+                let keyWindow = UIApplication.shared.getKeyWindow()
+                keyWindow?.addSubview(wkWebView)
+                UIView.animate(withDuration: 0.3, animations: {
+                    var webViewFrame = wkWebView.frame
+                    webViewFrame.origin.y = 0
+                    wkWebView.frame = webViewFrame
+                }, completion: nil)
+            }
+        }
+        self.listener?.onShow()
```

Comment:
Since all the above code is wrapped in an async block, the `self.listener?.onShow()` could potentially get called before the code within the block is executed. Is this intended? Just wanted to double check.